lib/transaction/trace/index.js

```diff
@@ -79,55 +77,55 @@ Trace.prototype.generateSpanEvents = function generateSpanEvents() {
   const config = this.transaction.agent.config
 
   const infiniteTracingConfigured = Boolean(config.infinite_tracing.trace_observer.host)
+  const sampledOrInfinite = infiniteTracingConfigured || this.transaction.sampled
+  const shouldProcess =
```

Comment:
I still find this hard to grok, since it's still essentially chained conditionals. My personal preference would be to abstract it into an function and we unit test the permutations.

I'd love for this to read like
```js
if (!shouldGenerateSpanEvents(config, this.transaction)) {
  return
}
```

Thoughts?

Reply:
That makes sense. There are a *lot* of conditionals in here.